### PR TITLE
Add Azure Cognitive Search vector connector

### PR DESCRIPTION
**Pull request by tawalke**
### Motivation and Context
  **Why is this change required?**
  Enable Azure Cognitive Search Vector functionality to Semantic Kernel
  
  **What problem does it solve?**
  Allow Semantic Kernel to leverage the Azure Cognitive Search vector search preview as connector

  **What scenario does it contribute to?**
  Persistent Memory connectors

### Description
Leverage the Azure Cognitive Search preview of vector search capabilities by using the Azure Python SDK

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by tawalke** [comment]
@farzad528 Please advise.

File: python/semantic_kernel/connectors/memory/azure_cog_search/acs_utils.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+) -> Union[AzureKeyCredential, DefaultAzureCredential, DefaultAzureCredentialSync]:
+    load_dotenv()
+    acs_key = os.getenv("AZURE_SEARCH_ADMIN_KEY")
+
+    if azsearch_api_key:
+        credential = (
+            DefaultAzureCredential() if use_async else DefaultAzureCredentialSync()
+        )
+    else:
+        credential = AzureKeyCredential(acs_key)
```
Comment (tot0) [inline python/semantic_kernel/connectors/memory/azure_cog_search/acs_utils.py:36]: `acs_key` could be `None` here (if `AZURE_SEARCH_ADMIN_KEY` not set in env), I think AzureKeyCredential will throw if that's the case, is that intended?

Comment (tawalke) [reply python/semantic_kernel/connectors/memory/azure_cog_search/acs_utils.py:36]: Thanks will review.

File: python/semantic_kernel/connectors/memory/azure_cog_search/acs_memory_store.py [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+
+        Returns:
+            MemoryRecord -- The record.
+        """
+
+        search_id = key
+
+        results = self._cogsearch_client.search(
+            search_text="",
+            vector=Vector(value=search_id, k=3, fields="collection_id"),
```
Comment (mattmsft) [inline python/semantic_kernel/connectors/memory/azure_cog_search/acs_memory_store.py:331]: why is k fixed to 3? shouldn't this be a parameter passed in through semantic kernel?

Comment (tawalke) [reply python/semantic_kernel/connectors/memory/azure_cog_search/acs_memory_store.py:331]: Thanks @mattmsft Making all the changes we discussed. You rock.

Comment (tawalke) [reply python/semantic_kernel/connectors/memory/azure_cog_search/acs_memory_store.py]: Changed to parameter passing in limit: https://github.com/microsoft/semantic-kernel/pull/1103/commits/ca595c56df60c1c0fa3c3adba706dc8e358886ea

**Comment by tawalke** [comment]
@mattmsft and @@farzad528: Made several updates from our conversations. Please review and advise as this is a Hot PR.

**Comment by tawalke** [comment]
@mattmsft ACS connector completion notes: 

The formatting and clean up commit to signal completion of the ACS connector for this PR was completed from yesterday. I noticed that the final commit to this PR is not reflective here for you. 

Please see commit: https://github.com/microsoft/semantic-kernel/commit/d749247d8e7fc1bd7d002b561cfeea5447da7b0e and feature branch: https://github.com/microsoft/semantic-kernel/tree/feature-azure-search

**Pull request merged**